blocks/feed-newsroom/feed-newsroom.js

```diff
@@ -0,0 +1,195 @@
+  await loadBlock(builtBlock);
+  builtBlock.before(filterDiv);
+  if (results.length > currentResults) {
+    builtBlock.after(loadMoreContainer);
+  } else loadMoreContainer.remove();
+
+  // In order to place fragment block to the right side of the page
+  const feedNewsroom = builtBlock.parentElement.parentElement.parentElement;
+  if (feedNewsroom.nextElementSibling.classList.contains('fragment-container')) {
+    feedNewsroom.nextElementSibling.lastChild.classList.add('others');
+    builtBlock.append(...feedNewsroom.nextElementSibling.childNodes);
+  }
+
+  return builtBlock;
+}
+
+// The Below function is leveraged for loading results when any year is selected in the dropdown
+async function loadYearResults(block, blockType, results, blockCfg) {
+  let slicedResults = 0;
+  const parentBlock = document.querySelector('.block.feed-newsroom > .others');
+  if (parentBlock.parentNode.nextElementSibling) parentBlock.parentNode.nextElementSibling.remove();
+  parentBlock.innerHTML = '';
+  slicedResults = results;
+  const blockContents = resultParsers[blockType](slicedResults, blockCfg);
+  const builtBlock = buildBlock(blockType, blockContents);
+
+  [...block.classList].forEach((item) => {
+    if (item !== 'feed') {
+      builtBlock.classList.add(item);
+    }
+  });
+  builtBlock.querySelectorAll(':scope > div').forEach((div) => {
+    div.classList.add('other');
+  });
+  parentBlock.append(...builtBlock.childNodes);
+}
+
+/**
+   * Feed block decorator to build feeds based on block configuration
+   */
+export default async function decorate(block) {
+  let searchResults = 0;
+  const chunk = 15;
+  const blockType = 'highlight';
+  const blockCfg = readBlockConfig(block);
+  console.log(`${getLanguage()}-search`);
+  const queryObj = await queryIndex(`${getLanguage()}-search`);
+
+  const omitPageTypes = getMetadataNullable('omit-page-types');
+  // eslint-disable-next-line prefer-arrow-callback
+  const results = queryObj.where(function filterElements(el) {
+    const elPageType = (el.pagetype ?? '').trim().toLowerCase();
+    let match = false;
+    match = (!omitPageTypes || !(omitPageTypes.split(',').includes(elPageType)));
+    return match;
+  })
+  // eslint-disable-next-line
+    .orderByDescending((el) => (blockCfg.sort ? parseInt(el[blockCfg.sort.trim().toLowerCase()], 10) : el.path))
+    .toList()
+    .filter((x) => { const itsDate = getFormattedDate(new Date(parseInt(x[blockCfg.sort.trim().toLowerCase()], 10))).split(', '); return (parseInt(itsDate[itsDate.length - 1], 10) > 2000); });
+  block.innerHTML = '';
+  // Creation of filter and year
+  const filterDiv = document.createElement('div');
+  filterDiv.innerHTML = `<form action="#">
+  <div class="filter-nav">
+    <span>
+      <select class="form-control" name="" id="news_year">
+        <option value="">Year</option>
+        </select>
+    </span>
+    <button data-nonce="8411f43402" data-post_type="news" " id="news_filter">FILTER</button>
```

Comment:
Needs localization.

Reply:
As seen in https://www.sunstar.com/jp/newsroom/ - there is no localisation for Filter, hence not did that. Please advise if still we need to do that .